easybuild/tools/systemtools.py

```diff
@@ -133,76 +103,107 @@ def get_core_count():
 
 
 def get_cpu_vendor():
-    """Try to detect the cpu identifier
+    """
+    Try to detect the CPU vendor
 
-    will return INTEL, ARM or AMD constant
+    @return: INTEL, ARM or AMD constant
     """
-    regexp = re.compile(r"^vendor_id\s+:\s*(?P<vendorid>\S+)\s*$", re.M)
-    VENDORS = {
-        'GenuineIntel': INTEL,
-        'AuthenticAMD': AMD,
-    }
+    vendor = None
     os_type = get_os_type()
 
-    if os_type == LINUX:
-        try:
-            txt = read_file('/proc/cpuinfo', log_error=False)
-            arch = UNKNOWN
-            # vendor_id might not be in the /proc/cpuinfo, so this might fail
-            res = regexp.search(txt)
-            if res:
-                arch = res.groupdict().get('vendorid', UNKNOWN)
-            if arch in VENDORS:
-                return VENDORS[arch]
-
-            # some embeded linux on arm behaves differently (e.g. raspbian)
-            regexp = re.compile(r"^Processor\s+:\s*(?P<vendorid>ARM\S+)\s*", re.M)
-            res = regexp.search(txt)
-            if res:
-                arch = res.groupdict().get('vendorid', UNKNOWN)
-            if ARM in arch:
-                return ARM
-        except IOError, err:
-            raise SystemToolsException("An error occured while determining CPU vendor since: %s" % err)
+    if os_type == LINUX and os.path.exists(PROC_CPUINFO_FP):
+        txt = read_file(PROC_CPUINFO_FP)
+        arch = UNKNOWN
+
+        vendor_regexes = [
+            r"^vendor_id\s+:\s*(\S+)\s*$",  # Linux/x86
+            r".*:\s*(ARM|IBM).*$",  # Linux/ARM (e.g., Raspbian), Linux/POWER
+        ]
+        vendor_regex = re.compile('|'.join(vendor_regexes), re.M)
+        res = vendor_regex.search(txt)
+        if res:
+            arch = res.group(1) or res.group(2)
```

Comment:
funny. try a named group (`(?P<cpu>\S+)` and `(?P<cpu>ARM|IBM)` and use `groupdict()['cpu']` here